alicloud/resource_alicloud_adb_cluster.go

```diff
@@ -0,0 +1,432 @@
+package alicloud
+
+import (
+	"strings"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/adb"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudAdbCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudAdbClusterCreate,
+		Read:   resourceAlicloudAdbClusterRead,
+		Update: resourceAlicloudAdbClusterUpdate,
+		Delete: resourceAlicloudAdbClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"db_cluster_version": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"3.0"}, false),
+				Optional:     true,
+				Default:      "3.0",
+			},
+			"db_cluster_category": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Basic", "Cluster"}, false),
+				Required:     true,
+			},
+			"db_cluster_network_type": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{strings.ToUpper(string(Vpc))}, false),
+				Default:      strings.ToUpper(string(Vpc)),
```

Comment:
I think db_cluster_network_type can be removed and it can be computed by whether setting `vswitch_id`.